File: C++/source/Framework/Messaging/LightHouse/LightHouseToolKit.cpp
/***************************************************************************
 *   Copyright (C) 20016 Vidyasagara Guntaka                               *
 *   All rights reserved.                                                  *
 *   Author : Vidyasagara Reddy Guntaka                                    *
 ***************************************************************************/

#include "Framework/Messaging/LightHouse/LightHouseToolKit.h"

namespace WaveNs
{

string LightHouseToolKit::s_lightHouseMulticastIpAddress = "239.255.23.16";
SI32   LightHouseToolKit::s_lightHouseMulticastPort      = 5011;

string LightHouseToolKit::getLightHouseMulticastIpAddress ()
{
    return (s_lightHouseMulticastIpAddress);
}

void LightHouseToolKit::setLightHouseMulticastIpAddress (const string &lightHouseMulticastIpAddress)
{
    s_lightHouseMulticastIpAddress = lightHouseMulticastIpAddress;
}

SI32 LightHouseToolKit::getLightHouseMulticastPort ()
{
    return (s_lightHouseMulticastPort);
}

void LightHouseToolKit::setLightHouseMulticastPort (const SI32 &LightHouseMulticastPort)
{
    s_lightHouseMulticastPort = LightHouseMulticastPort;
}

}

File: Java/source/com/CxWave/Wave/Framework/Messaging/LightHouse/LightHouseToolKit.java
/*************************************************************************************************************************
 * Copyright (C) 2015-2016 Vidyasagara Guntaka & CxWave, Inc * All rights reserved. * Author : Vidyasagara Reddy Guntaka *
 *************************************************************************************************************************/

package com.CxWave.Wave.Framework.Messaging.LightHouse;

public class LightHouseToolKit
{
    private static String m_lightHouseMulticastIpAddress = "239.255.23.16";
    private static int    m_lightHouseMulticastPort      = 5011;

    private LightHouseToolKit ()
    {
    }

    public static String getLightHouseMulticastIpAddress ()
    {
        return m_lightHouseMulticastIpAddress;
    }

    public static void setLightHouseMulticastIpAddress (final String lightHouseMulticastIpAddress)
    {
        m_lightHouseMulticastIpAddress = lightHouseMulticastIpAddress;
    }

    public static int getLightHouseMulticastPort ()
    {
        return m_lightHouseMulticastPort;
    }

    public static void setLightHouseMulticastPort (final int lightHouseMulticastPort)
    {
        m_lightHouseMulticastPort = lightHouseMulticastPort;
    }
}
